### PR TITLE
Warn player about cost of implant uplink

### PR DESCRIPTION
**Pull request by celotajstg**
## About The Pull Request

This PR adds a note regarding the implant uplink's price to the game preferences.
image
image

## Why It's Good For The Game

I used to think the implant uplink was free if it is on roundstart because it was not written anywhere it takes 4 TC. This is confusing.

## Changelog
:cl: Celotajs
qol: You can now see in game preferences that the implant uplink takes 4 TC from you.
/:cl:

### CODE REVIEW

File: code/_globalvars/lists/flavor_misc.dm [hunk 1]
```diff
@@ -173,7 +173,7 @@ GLOBAL_LIST_INIT(jumpsuitlist, list(PREF_SUIT, PREF_SKIRT))
 #define UPLINK_PDA "PDA"
 #define UPLINK_RADIO "Radio"
 #define UPLINK_PEN "Pen" //like a real spy!
-#define UPLINK_IMPLANT  "Implant"
+#define UPLINK_IMPLANT  "Implant (-4 TC)"
```
Comment (MrDoomBringer) [inline code/_globalvars/lists/flavor_misc.dm:176]: This should use the actual cost of the uplink implant. Not certain if you want to handle that behavior in this define though..

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/_globalvars/lists/flavor_misc.dm [hunk 2]
```diff
@@ -173,8 +173,12 @@ GLOBAL_LIST_INIT(jumpsuitlist, list(PREF_SUIT, PREF_SKIRT))
 #define UPLINK_PDA "PDA"
 #define UPLINK_RADIO "Radio"
 #define UPLINK_PEN "Pen" //like a real spy!
-#define UPLINK_IMPLANT  "Implant"
-GLOBAL_LIST_INIT(uplink_spawn_loc_list, list(UPLINK_PDA, UPLINK_RADIO, UPLINK_PEN, UPLINK_IMPLANT))
+#define UPLINK_IMPLANT "Implant"
+#define UPLINK_IMPLANT_WITH_PRICE "[UPLINK_IMPLANT] (-[UPLINK_IMPLANT_TELECRYSTAL_COST] TC)"
+// What we show to the user
+GLOBAL_LIST_INIT(uplink_spawn_loc_list, list(UPLINK_PDA, UPLINK_RADIO, UPLINK_PEN, UPLINK_IMPLANT_WITH_PRICE))
```
Comment (AnturK) [inline code/_globalvars/lists/flavor_misc.dm:179]: You're not using this list. Could also make this single associative list with saved_value = description

Comment (celotajstg) [reply code/_globalvars/lists/flavor_misc.dm:179]: This list is used on line 1508 in preferences.dm: https://github.com/tgstation/tgstation/pull/58872/files#diff-abde2422921045156657700c33fd411dd0029a7800d7892ccff2b5923ebc5132R1508

Should I remove it from being a global list and turn it into a list() on that line?

Comment (celotajstg) [reply code/_globalvars/lists/flavor_misc.dm:179]: Also tried to consider the associative list idea - wouldn't really work due to line 490 in preferences_savefile.dm https://github.com/tgstation/tgstation/pull/58872/files#diff-5705df368d810d2ef8e755ea140098cb348b8e1119c392b196f1ed997fee2729R490

Comment (Mothblocks) [reply code/_globalvars/lists/flavor_misc.dm:179]: i think the diff confused him into thinking this is a new list

File: code/_globalvars/lists/flavor_misc.dm [hunk 3]
```diff
@@ -173,8 +173,12 @@ GLOBAL_LIST_INIT(jumpsuitlist, list(PREF_SUIT, PREF_SKIRT))
 #define UPLINK_PDA "PDA"
 #define UPLINK_RADIO "Radio"
 #define UPLINK_PEN "Pen" //like a real spy!
-#define UPLINK_IMPLANT  "Implant"
-GLOBAL_LIST_INIT(uplink_spawn_loc_list, list(UPLINK_PDA, UPLINK_RADIO, UPLINK_PEN, UPLINK_IMPLANT))
+#define UPLINK_IMPLANT "Implant"
+#define UPLINK_IMPLANT_WITH_PRICE "[UPLINK_IMPLANT] (-[UPLINK_IMPLANT_TELECRYSTAL_COST] TC)"
```
Comment (Mothblocks) [inline code/_globalvars/lists/flavor_misc.dm:177]: This is a horrible way of doing it, now if we ever change the price we have to factor in that it'll fuck with people's data.

Keep the keys the same, make the input ITSELF display something different.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
You're doing it badly.

File: code/_globalvars/lists/flavor_misc.dm (hunk 3 above)
Comment (celotajstg) [reply code/_globalvars/lists/flavor_misc.dm:177]: No, the whole idea is that it will NOT mess with people's data. Only "Implant" is saved.

**Comment by Mothblocks** [comment]
Nevermind.

**Comment by MrDoomBringer** [comment]
contributor for 1 month vs two of our best maintainers
underdog story

**Pull request merged**